### ISSUE TITLE
The mAP and loss value is too high from first epoch

### ISSUE THREAD

**Issue by NevilleMthw** [keras-team/keras#16948]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu
- TensorFlow version (use command below): 2.4.1
- Python version: 3.8.0
- GPU model and memory: NVIDIA GeForce RTX

**Describe the problem**.

The problem I am facing is with my mean average precision value is too perfect and my loss value is too high from the beginning of my model training which is quite weird and I cannot seem to understand the issue. I am doing an object detection task with box coordinates, I seem to think the labels have something to do with the absurd result or the calculation of mAP.

**Describe the current behavior**.

The current behaviour occurs when I try to train my model using ```model.fit()``` function and then I receive the absurd results. The mAP result is 1.000 and loss is too high.

**Describe the expected behavior**.

The expected behaviour would be to have an accurate result and the correct loss value at the current model configuration.

**Standalone code to reproduce the issue**.

I have posted the code on my GitHub repo which is this specific notebook (the results are also shown): https://github.com/NevilleMthw/Stenosis-Project/blob/master/functional.ipynb
The problem is during the ```model.fit()``` command where I can see the inaccurate results.
```
model.compile(loss='mse', optimizer='adam', metrics=[tfr.keras.metrics.MeanAveragePrecisionMetric()])
model_history = model.fit(dataGenerator.flow(train_images, train_targets, batch_size=32),
                            validation_data=(val_images, val_targets), 
                            epochs=250, callbacks=[es], verbose=1, shuffle=True)
```

Before the training, I also read all the images and labels accordingly with this part of the code:
```
for index, row in test_labels_csv.iterrows():
    (filename, width, height, class_name, xmin, ymin, xmax, ymax) = row
    pic = cv2.imread(os.path.join(test_path,filename))

    x = pic.shape[0]
    y = pic.shape[1]
    
    img_Target_size = 224
    x_scale = img_Target_size / x
    y_scale = img_Target_size / y

    pic = cv2.resize(pic, (img_Target_size, img_Target_size))
    
    pic = pic / x

    xmin = float(xmin * x_scale)
    ymin = float(ymin * y_scale)
    xmax = float(xmax * x_scale)
    ymax = float(ymax * y_scale)

    test_images.append(pic)
    test_targets.append((xmin, ymin, xmax, ymax))
```
**Source code / logs**.
```
188/188 [==============================] - 33s 122ms/step - loss: 3480.4146 - mean_average_precision_metric: 1.0000 - val_loss: 1990.5461 - val_mean_average_precision_metric: 1.0000
Epoch 2/250
188/188 [==============================] - 20s 107ms/step - loss: 1341.6710 - mean_average_precision_metric: 1.0000 - val_loss: 2417.2395 - val_mean_average_precision_metric: 1.0000
Epoch 3/250
188/188 [==============================] - 20s 107ms/step - loss: 1226.9118 - mean_average_precision_metric: 1.0000 - val_loss: 1300.4974 - val_mean_average_precision_metric: 1.0000
```

Hopefully the issue is clear and understood. Appreciate any help.

**Comment by tilakrayal** [comment]
@NevilleMthw,
Tensorflow version 2.4 not actively supported. Hence, kindly update to the latest stable version 2.9 and let us know if you are facing the same issue. Thank you!

**Comment by NevilleMthw** [comment]
Hi @tilakrayal, I have updated my version TF 2.8 and now the model seems to function. But now there is a different issue which shows the mAP value as an error:
```
183/183 [==============================] - 13s 53ms/step - loss: 0.4941 - mean_io_u: 1.0000 - mean_average_precision_metric: 0.0000e+00 - val_loss: 0.0224 - val_mean_io_u: 1.0000 - val_mean_average_precision_metric: 0.0000e+00
Epoch 2/250
183/183 [==============================] - 7s 36ms/step - loss: 0.0225 - mean_io_u: 1.0000 - mean_average_precision_metric: 0.0000e+00 - val_loss: 0.0223 - val_mean_io_u: 1.0000 - val_mean_average_precision_metric: 0.0000e+00
```
No matter the model, I still seem to get this value. I have also checked my data which is normalized and scaled. I am using 'mse' as my loss function with 'adam' as the optimizer.

**Comment by tilakrayal** [comment]
@NevilleMthw,
I was not able to download the files attached. Could you please provide the access for the required files to download which helps to reproduce the issue in local. Thank you!

**Comment by NevilleMthw** [comment]
Hi @tilakrayal,

I have solved the issue by using another metric to calculate the results. Thank you for your time!
